### PR TITLE
feat: pangoro faucet

### PR DESCRIPTION
**Pull request by cinlk**
ORING token faucet

### CODE REVIEW

File: airdrop/api/pangoro/airdrop/state.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      message: `Sorry, we can't find your ip address`,
+    };
+    res.end(JSON.stringify(body, null, 2));
+    return;
+  }
+
+  const chainName = 'PANGORO';
+  const cacheKeyIp = `${chainName}-${user.id}-${ip}`;
+
+  const recordTime = await client.get(cacheKeyIp);
```
Comment (sxlwar) [inline airdrop/api/pangoro/airdrop/state.ts:62]: always use !== or ===

 When == or != must be used, add a comment

File: airdrop/api/pangoro/airdrop/transfer.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  if (!ip) {
+    res.statusCode = 403;
+    const body = {
+      err: 1,
+      message: `Sorry, we can't find your ip address.`,
+    };
+    res.end(JSON.stringify(body, null, 2));
+    return;
+  }
+  // const
```
Comment (sxlwar) [inline airdrop/api/pangoro/airdrop/transfer.ts]: remove useless comment

File: airdrop/api/pangoro/airdrop/transfer.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+  const data = qs.parse(req.body);
+
+  res.setHeader('content-type', 'application/json');
+
+  // check data
+  if (!data || typeof data.address != 'string') {
```
Comment (sxlwar) [inline airdrop/api/pangoro/airdrop/transfer.ts]: !=

**Review by sxlwar — CHANGES_REQUESTED** [review]

**Comment by sxlwar** [comment]
建议不要忽略 lint，可以避免语法上可能导致的 bug

File: airdrop/api/grant.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+function getConfig() {
+  const config = {
+    defaults: {
+      origin: 'https://docs.darwinia.network',
```
Comment (fewensa) [inline airdrop/api/grant.ts]: Wrong origin?

**Comment by cinlk** [comment]
refactor in #284

**Pull request closed without merging**